subsys/net/ip/net_context.c

```diff
@@ -399,6 +427,17 @@ static int bind_default(struct net_context *context)
 	}
 #endif
 
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_PACKET) && family == AF_PACKET) {
+		struct sockaddr_ll ll_addr;
+
+		ll_addr.sll_family = AF_PACKET;
+		ll_addr.sll_protocol = ETH_P_ALL;
+		ll_addr.sll_ifindex = net_if_get_by_iface(net_if_get_default());
```

Comment:
Do you remember, that in POSIX, interface index can't be 0 (zero means "doesn't care"). That effectively means that indexes start from 1. So, this is not correct.

Reply:
> Do you remember, that in POSIX, interface index can't be 0 (zero means "doesn't care").

Don't know about that.

Reply:
> That effectively means that indexes start from 1. So, this is not correct.

i.e. we have to change our indexing from 1.

Reply:
> Don't know about that.

Well, please read up on that in http://man7.org/linux/man-pages/man7/packet.7.html :

> sll_ifindex is the interface index of the interface (see
          netdevice(7)); 0 matches any interface (only permitted for bind‐
          ing).  sll_hatype is an ARP type as defined in the
          <linux/if_arp.h> include file.


---
> i.e. we have to change our indexing from 1.

Yes, we would need to +1 internal Zephyr number when we pass it to *socket* functions, and -1 what we get from user via *socket* functions. I'm just not sure at what level to do that, but given that we *have* to do that, butter just do that in net_context functions, to avoid having different indexes in different API. @jukkar, @tbursztyka,  maybe you'd have suggestion here.

Reply:
I noticed the same thing that linux CAN manual pages talk about index 0 being special. Not sure how we could accomplish that currently, it sounds very error prone if we start to +-1 the interface index value in net_context. Probably the best option is to make net_if_get_by_index() do the -1 and net_if_get_by_iface() to return +1 for the network interface index value and make sure that no code accesses network interface array directly. I would leave this issue after 1.14 merge window closes.

Reply:
> Probably the best option is to make net_if_get_by_index() do the -1 and net_if_get_by_iface() to return +1 for the network interface index value and make sure that no code accesses network interface array directly.

I can only +1 such a solution: it's arguably a wart, I don't like 1-based indexing myself, but if POSIX specifies that (and likely it's not in POSIX explicitly, it's just how BSD did it, and all UNIXes adhere to), it's better to adopt that scheme as native to Zephyr too.

While it's not pretty formal decision, pragmatically I agree that we probably could live with this "numbering bug" for some time (i.e. not do anything in this patch), if we don't forget to fix it afterwards.

Reply:
> Is there anything to do further? shall I mark this conversation resolved?

I ticketed that as https://github.com/zephyrproject-rtos/zephyr/issues/13084, and in the circumstances we have, I agree it's "ok" to keep stuff as is here.

Reply:
Ok, thank you.